+nosnoc/+discrete_time_problem/Stewart.m

```diff
@@ -908,8 +919,141 @@ function create_solver(obj, solver_options, plugin)
                     end
                 end
             end
+            
+            if ~isempty(active_set)
+                [IG,IH,~] = obj.process_active_set(active_set);
+            else
+                IG = [];
+                IH = [];
+            end
+            stats = solve@vdx.problems.Mpcc(obj, IG=IG, IH=IH);
+        end
+
+        function [IG,IH,I00] = process_active_set(obj, active_set)
+        % This method takes a nosnoc active set for a PSS and produces an active set for the
+        % complementarity constraints in this problem. It returns these as a boolean array.
+        %
+        % TODO(@anton) there should be a cleaner way of doing this but this is "easy".
```

Comment:
what do you mean

Reply:
I apply the active set directly to the algebraic values then get the MPCC active set by evaluating $G$ and $H$. In principle we could  generate the MPCC active set "directly" with some indexing manipulations.

I am less convinced now that this is actually that bad as it is less likely to be broken by some changes to ordering etc. Perhaps I drop the TODO.